### PR TITLE
LOOP-1650: fix crash with two pickers scrolling in opposite directions at the same time

### PR DESCRIPTION
**Pull request by novalegra**
This error is occurring because the picker gets into a state where the “low” bound of the range is actually greater than the “high” bound of the range. I changed it so that if (and only if) the picker runs into this situation, it picks the higher of the two values (ex: if picker thinks it’s 110-106 mg/dL, it’ll change to be 110-110 mg/dL).

https://tidepool.atlassian.net/browse/LOOP-1650

### CODE REVIEW

File: LoopKitUI/Views/GlucoseRangePicker.swift [hunk 1]
```diff
@@ -39,11 +39,26 @@ public struct GlucoseRangePicker: View {
...
+                    // Prevent crash if picker gets into state where "lower bound" > "upper bound"
+                    range.wrappedValue = $0...$0
+                }
+                range.wrappedValue = $0...range.wrappedValue.upperBound
+                
+        }
         )
         self._upperBound = Binding(
             get: { range.wrappedValue.upperBound },
-            set: { range.wrappedValue = range.wrappedValue.lowerBound...$0 }
```
Comment (rickpasetto) [inline LoopKitUI/Views/GlucoseRangePicker.swift:46]: [nit] seems generally useful.  Might it make sense to add a utility function like this?
```
func safeClosedRange(minimum: Int, maximum: Int) -> ClosedRange<Int> {
    if minimum > maximum { return minimum...minimum }
    return minimum...maximum
}
```
?

**Review by rickpasetto — APPROVED** [review]
LGTM (with one minor suggestion)

File: LoopKitUI/Views/GlucoseRangePicker.swift (hunk 1 above)
Comment (ps2) [reply LoopKitUI/Views/GlucoseRangePicker.swift:46]: Would need either two functions `closedRangeBoundedByMinimum` and `closedRangeBoundedByMaximum` or a method that takes an additional argument that specifies the type of resolution preferred (min or max).  Those don't sound worth it, unless this appears as a common pattern in the code.

**Review by ps2 — APPROVED** [review]
LGTM!

File: LoopKitUI/Views/GlucoseRangePicker.swift (hunk 1 above)
Comment (novalegra) [reply LoopKitUI/Views/GlucoseRangePicker.swift:46]: This is only the case in this one place, so I'm not convinced we need it (happy to revisit this later if we find more bugs like this)

**Pull request merged**